### ISSUE TITLE
User-specified offset

### ISSUE THREAD

**Issue by Mic92** [sharkdp/hexyl#57]
I would like to integrate hexyl in GDB using the python API. Therefore it would be useful if I could supply a different offset, so that input offsets in hexyl matches with the addresses in virtual memory in the process.

**Comment by Mic92** [comment]
This is what I currently came up with:


```python
class Hexyl(gdb.Command):
    """
       Run hexyl on provided symbol/address. Takes the number of bytes to print as an optional parameter
    """
    def __init__(self):
        super(Hexyl, self).__init__("hexyil", gdb.COMMAND_USER)
        self.long_int = gdb.lookup_type("long")

    def invoke(self, arg, from_tty):
        argv = gdb.string_to_argv(arg)
        count = 0x100
        if argv and len(argv) > 0:
            address = argv[0]
            if len(argv) > 1:
                count = argv[1]
        else:
            address = "$sp"

        with tempfile.NamedTemporaryFile(suffix='.bin', delete=True) as f:
            gdb.execute("dump binary memory %s %s (%s + %s)" % (f.name, address, address, count))
            subprocess.call(["hexyl", f.name])

        return False
```

**Comment by sharkdp** [comment]
Thank you very much for your feedback! Integrating `hexyl` in `gdb` sounds like a very cool idea!

Can you go into a little bit more detail what kind of feature you would like to see in `hexyl` to make this easier?

**Comment by Mic92** [comment]
A parameter that allows to add an offset to the number reported by hexyl on the left-side would be sufficient. This way the addresses by gdb would match up with hexyl offsets.

**Comment by sharkdp** [comment]
Ok, so this would be similar to `xxd`s `-o <offset>` option. Sounds good to me!

**Comment by matematikaadit** [comment]
So, if I understand it correctly, this is just a display option, right? i.e. hexyl still reads the binary from the start of the file, but it display them as if the file is started from a certain virtual address.

For example:

File content: `[ 7f 45 4c 46 02 01 01 00 ]`

```
$ hexyl foo
┌────────┬─────────────────────────┬─────────────────────────┬────────┬────────┐
│00000000│ 7f 45 4c 46 02 01 01 00 ┊                         │•ELF•••0┊        │
└────────┴─────────────────────────┴─────────────────────────┴────────┴────────┘
$ hexyl --display-offset 0x408000 foo  # just for example
┌────────┬─────────────────────────┬─────────────────────────┬────────┬────────┐
│00408000│ 7f 45 4c 46 02 01 01 00 ┊                         │•ELF•••0┊        │
└────────┴─────────────────────────┴─────────────────────────┴────────┴────────┘
```

**Comment by sharkdp** [comment]
closed via #72 by @tommilligan
